pkg/addons/adminconsole/adminconsole.go

```diff
@@ -86,81 +104,156 @@ func (a *AdminConsole) addLicenseToHelmValues() error {
 	return nil
 }
 
-func (a *AdminConsole) Apply(ctx context.Context) error {
-	version, err := a.Latest()
-	if err != nil {
-		return fmt.Errorf("unable to get latest Admin Console version: %w", err)
+// GetPasswordFromConfig returns the adminconsole password from the cluster config.
+func (a *AdminConsole) GetPasswordFromConfig() (string, error) {
+
+	chartList := a.config.Spec.Extensions.Helm.Charts
+
+	pass := ""
+
+	for _, chart := range chartList {
+		if chart.Name == "adminconsole" {
+			values := dig.Mapping{}
+			err := yaml.Unmarshal([]byte(chart.Values), &values)
+			if err != nil {
+				return "", fmt.Errorf("unable to unmarshal values: %w", err)
+			}
+			pass = values["password"].(string)
+			break
+		}
 	}
-	if !semver.IsValid(version) {
-		return fmt.Errorf("unable to parse version %s", version)
+
+	if pass == "" {
+		return "", fmt.Errorf("unable to find adminconsole chart in cluster config")
+	} else {
```

Comment:
This else is not necessary. It just creates a warning on my linter:

```
if block ends with a return statement, so drop this else and outdent its block
```

I think it would be better and cleaner to just return empty if a password could not be find (judging by how you are using the values returned by this function).